session/pingpong/hermes_promise_handler.go

```diff
@@ -332,6 +332,9 @@ func (aph *HermesPromiseHandler) requestPromise(er enqueuedRequest) {
 	})
 
 	err = aph.revealR(ap)
+	if err != nil {
```

Comment:
this looks like its making the next function useless

Reply:
I guess the log should be in the handle func, good catch.